src/django_mysql/rewrite_query.py

```diff
@@ -187,4 +194,17 @@ def modify_sql_index_hints(sql, table_name, rule, index_names, for_what):
         for_section=for_section,
         index_names=("" if index_names == "NONE" else index_names),
     )
-    return re.sub(table_spec_re, replacement, sql, count=1, flags=re.VERBOSE)
+    matches = [
+        match
+        for match in re.finditer(table_spec_re, sql, flags=re.VERBOSE)
+        if match.start(0) < start_pos
+        # only need to look at the position before inserted comment
+    ]
+    nearest_match_start = matches[len(matches) - 1].start(0)
+
+    new_sql = sql[0:nearest_match_start] + re.sub(table_spec_re, replacement, sql[nearest_match_start:], count=1, flags=re.VERBOSE)
```

Comment:
nit:

`sql[0:nearest_match_start]` can be simplify to `sql[:nearest_match_start]`